src/streams.cpp

```diff
@@ -5,6 +5,54 @@
 #include <span.h>
 #include <streams.h>
 
+#include <array>
+
+std::size_t XorFile::detail_fread(Span<std::byte> dst)
+{
+    if (!m_file) throw std::ios_base::failure("XorFile::read: file handle is nullptr");
+    const auto init_pos{std::ftell(m_file)};
+    if (init_pos < 0) return 0;
```

Comment:
fa9325a14584be1c5aa6b4cdefbdaa69b27e402d

Why not throw a `std::ios_base::failure()` here? From what I can tell, we'd never expect to get back a negative error value.